File: TP 1/c/aquere.c
#include <stdio.h>
#include <string.h>

int letrama(char pala[],int i){
    int tam=0;
    int resp=0;
    tam=strlen(pala);

    if (i < tam) {
        if (pala[i] >= 'A' && pala[i] <= 'Z') {
            resp = resp +1;
        }
        resp += letrama(pala, i + 1);
    }
    
    return resp;
}


void main(){
    char pala[500]="";
    int resp=0;
    int i=0;
    scanf("%[^\n]", pala);
    getchar();
    while(strcmp(pala, "FIM")!=0){
        resp=letrama(pala, i);
        printf("%d\n", resp);
        scanf(" %[^\n]", pala);
        getchar();

    }



}

File: TP 1/java/aquere.java
import java.util.Scanner;

public class aquere {
    
    static int letrama(String pala, int i) {
        int resp = 0;
        if(i < pala.length()){
            if (pala.charAt(i) >= 'A' && pala.charAt(i) <= 'Z') {
                    resp++;
            }
            resp=resp + letrama(pala, i+1);
        }
        return resp;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String pala = "";  
        int i=0;
        int resp = 0;

        pala = sc.nextLine();
        while (!pala.equals("FIM")) {
            resp = letrama(pala, i);
            System.out.println(resp);
            pala = sc.nextLine();
        }

        sc.close(); 
    }
}
